test/functional/tests/volumes/test_volume_partitions.py

```diff
@@ -0,0 +1,177 @@
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_cas_preserves_partitions(partition_table, filesystem, cache_mode):
+    """
+        title: Volume test for preserving partition table from core device.
+        description: |
+          Validation of the ability of CAS to preserve partition table on core device
+          after adding it to cache.
+        pass_criteria:
+          - Md5 sums on partitions shall be identical before and after running cache.
+          - Partition table shall be preserved on exported object.
+    """
+    with TestRun.step(f"Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_size = []
+        for i in range(cores_number):
+            core_size.append(Size(1, Unit.GibiByte))
+        core_dev.create_partitions(core_size, partition_table)
+
+    with TestRun.step("Create filesystem on core devices."):
+        for i in range(cores_number):
+            core_dev.partitions[i].create_filesystem(filesystem)
+
+    with TestRun.step("Mount core devices and create test files."):
+        files = []
+        for i, core in enumerate(core_dev.partitions):
+            mount_path = f"{mount_point}{i}"
+            core.mount(mount_path)
+            test_file_path = f"{mount_path}/test_file"
+            files.append(fs_utils.create_test_file(test_file_path))
+
+    with TestRun.step("Check md5 sums of test files."):
+        test_files_md5sums = []
+        for file in files:
+            test_files_md5sums.append(file.md5sum())
+
+    with TestRun.step("Unmount core devices."):
+        for core in core_dev.partitions:
+            core.unmount()
+
+    with TestRun.step(f"Start cache."):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+
+    with TestRun.step("Add cores to cache."):
+        cores = []
+        for i in range(cores_number):
+            cores.append(cache.add_core(core_dev.partitions[i]))
+
+    with TestRun.step("Mount core devices."):
+        for i, core in enumerate(cores):
+            mount_path = f"{mount_point}{i}"
+            core.mount(mount_path)
+
+    with TestRun.step("Check again md5 sums of test files."):
+        test_files_md5sums_new = []
+        for file in files:
+            test_files_md5sums_new.append(file.md5sum())
+
+    with TestRun.step("Unmount core devices."):
+        for core in cores:
+            core.unmount()
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Compare md5 sums of test files."):
+        if test_files_md5sums != test_files_md5sums_new:
+            TestRun.LOGGER.error("Md5 sums are different.")
```

Comment:
TestRun. fail() because there is no reason for this test to execute any other things after that